File: tasks/task03/Prime.c
#include <stdio.h>
#include <stdbool.h>

bool is6k1(int num) {
    if (num <= 1) return false;
    if (num <= 3) return true;
    if (num % 2 == 0 || num % 3 == 0) return false;

    int i = 5;
    while (i * i <= num) {
        if (num % i == 0 || num % (i + 2) == 0) return false;
        i += 6;
    }
    return true;
}

int main() {
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);
    printf("Prime till %d are:\n", n);
    for (int i = 2; i <= n; i++) {
        if (is6k1(i)) printf("%d\n", i);
    }
    return 0;
}

File: tasks/task03/Prime.exs
defmodule Prime do
  def is_6k_1(num) do
    cond do
      num <= 1 -> false
      num <= 3 -> true
      rem(num, 2) == 0 or rem(num, 3) == 0 -> false
      true -> do_check(num, 5)
    end
  end

  defp do_check(num, i) when i * i > num, do: true
  defp do_check(num, i) when rem(num, i) == 0 or rem(num, i + 2) == 0, do: false
  defp do_check(num, i) do
    do_check(num, i + 6)
  end
end

IO.puts "Enter a number: "
n = String.trim(IO.gets("")) |> String.to_integer()
IO.puts "Prime till#{n} are:"
Enum.each(2..n, fn i -> if Prime.is_6k_1(i), do: IO.puts(i) end)
